### PR TITLE
chore(CI): refactor yarn `smoke-test`

### PR DESCRIPTION
**Pull request by jtoar**
There's a few footguns to running smoke tests locally (and if you have to run smoke tests locally, it's already painful enough):
- you don't have a test project, or you're running against a different test project than you expected
- `yarn rwfw project:sync` isn't running
- you haven't built the project before running the prerender or serve smoke test
- etc.

The `yarn smoke-test` script as it was checked for some of these things, but in an incomplete way that could lead to more needless frustration. For example, it would check if you had an env var set, but not if the test project actually existed at that env var. It didn't handle passing flags to the underlying playwright command (`npx playwright test`) too well. This polishes things up a bit:

- it used to be `yarn smoke-test` (no "s")—changed to `yarn smoke-tests`. a potentially annoying change in the short term, but i wanted it to be more correct—it's for running more than one, and the dir is called `smoke-tests`, etc.
- adds a little `README.md` that explains some basics about what the directory is for

- better help text

image

- if you don't provide a test to run, it prompts you:

image

- you can select more than one

image

- warns if `yarn rwfw project:sync` isn't running

image

- flags actually pass through to playwright

image

### CODE REVIEW

**Comment by jtoar** [comment]
@Tobbe requested you for review because you worked with me on this a while ago I think

File: tasks/smoke-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# Smoke tests
+
+These smoke tests run against the test project fixture (./\_\_fixtures\_\_/test-project).
```
[the remark this answers is not in the document]
Comment (jtoar) [reply tasks/smoke-tests/README.md:3]: I had it this way at first but markdown tried to italicize it when it was that way so I don't think it'll work for the rendered version of the README or for making an actual link to a file path.

File: tasks/smoke-tests/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+REDWOOD_TEST_PROJECT_PATH=<test project path>
+```
+
+Redwood has a mini CLI for running the smoke tests:
+
+```
+yarn smoke-tests [options] [smoke-test..]
+```
+
+If you don't specify any smoke tests to run, it'll prompt you:
```
[the remark this answers is not in the document]
Comment (jtoar) [reply tasks/smoke-tests/README.md:26]: Yeah good call

File: tasks/smoke-tests/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jtoar) [reply tasks/smoke-tests/README.md:3]: Looks like I forgot to make this a link

File: tasks/smoke-tests/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  - remove files that were written or undo file changes. (for this reason, you may find it helpful to initialize a git repo in the test project before running smoke tests)
+
+- `yarn rwfw project:sync` needs to be running in the test project to test framework changes
+  - if you created the test project with `--link`, it should already be running. if not, just run it in the test project
+
+- some smoke tests, like prerender or serve, need the test project to be built (`yarn rw build`)
+
+The CLI tries to warn you about all of these (but the first; that'll have to be fixed by refactoring the tests).
+
+## Adding a smoke test category
```
[the remark this answers is not in the document]
Comment (jtoar) [reply tasks/smoke-tests/README.md:61]: Yeah don't mind much either way

File: tasks/smoke-tests/README.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+```
+cd ./tasks/smoke-tests/dev
+npx playwright test
+```
+
+### Gotchas
+
+There's a few gotchas to be aware of:
+
+- not all smoke tests are idempotent. this means you may need to...
```
[the remark this answers is not in the document]
Comment (jtoar) [reply tasks/smoke-tests/README.md:50]: I may be taking some liberties here, I'll rework it

File: tasks/smoke-tests/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jtoar) [reply tasks/smoke-tests/README.md]: Addressed in https://github.com/redwoodjs/redwood/pull/9381/commits/bda35ff8971e9a64737ebf9254ce9d490d8e2110

File: tasks/smoke-tests/README.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  - reset the test project's database (`yarn rw prisma migreate reset --force`) between runs
+  - remove files that were written or undo file changes. (for this reason, you may find it helpful to initialize a git repo in the test project before running smoke tests)
+
+- `yarn rwfw project:sync` needs to be running in the test project to test framework changes
+  - if you created the test project with `--link`, it should already be running. if not, just run it in the test project
```
[the remark this answers is not in the document]
Comment (jtoar) [reply tasks/smoke-tests/README.md]: Addressed in https://github.com/redwoodjs/redwood/pull/9381/commits/bda35ff8971e9a64737ebf9254ce9d490d8e2110

File: tasks/smoke-tests/README.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+npx playwright test
+```
+
+### Gotchas
+
+There's a few gotchas to be aware of:
+
+- not all smoke tests are idempotent. this means you may need to...
+  - reset the test project's database (`yarn rw prisma migreate reset --force`) between runs
+  - remove files that were written or undo file changes. (for this reason, you may find it helpful to initialize a git repo in the test project before running smoke tests)
```
[the remark this answers is not in the document]
Comment (jtoar) [reply tasks/smoke-tests/README.md]: Addressed in https://github.com/redwoodjs/redwood/pull/9381/commits/bda35ff8971e9a64737ebf9254ce9d490d8e2110

File: tasks/smoke-tests/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jtoar) [reply tasks/smoke-tests/README.md]: Addressed in https://github.com/redwoodjs/redwood/pull/9381/commits/bda35ff8971e9a64737ebf9254ce9d490d8e2110

File: tasks/smoke-tests/README.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+```
+yarn smoke-tests [options] [smoke-test..]
+```
+
+If you don't specify any smoke tests to run, it'll prompt you:
+
+Smoke tests are written using [Playwright](https://playwright.dev/).
+You can pass options to the underlying Playwright test command (`npx playwright test`) using `--playwrightOptions`.
+Some common ones are `--debug` (to step through the test) and `--headed` (to see what's happening):
```
[the remark this answers is not in the document]
Comment (jtoar) [reply tasks/smoke-tests/README.md]: Addressed in https://github.com/redwoodjs/redwood/pull/9381/commits/bda35ff8971e9a64737ebf9254ce9d490d8e2110

File: tasks/smoke-tests/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jtoar) [reply tasks/smoke-tests/README.md]: Addressed in https://github.com/redwoodjs/redwood/pull/9381/commits/bda35ff8971e9a64737ebf9254ce9d490d8e2110

File: tasks/smoke-tests/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jtoar) [reply tasks/smoke-tests/README.md]: Updated in https://github.com/redwoodjs/redwood/pull/9381/commits/bda35ff8971e9a64737ebf9254ce9d490d8e2110.

It's the correct link now. Clicking it should focus the directory in your IDE. In GitHub, it should take you to the directory.

File: tasks/smoke-tests/README.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+npx playwright test
+```
+
+### Gotchas
+
+There's a few gotchas to be aware of:
+
+- Not all smoke tests are idempotent
+
+  This means you may need to reset the test project's database (`yarn rw prisma migreate reset --force`) between runs or remove files that were written or undo file changes. (For this reason, you may find it helpful to initialize a git repo in the test project before running smoke tests.)
```
Comment (Tobbe) [inline tasks/smoke-tests/README.md:53]:
```suggestion
  This means you may need to reset the test project's database (`yarn rw prisma migrate reset --force`) between runs or remove files that were written or undo file changes. (For this reason, you may find it helpful to initialize a git repo in the test project before running smoke tests.)
```

File: tasks/smoke-tests/README.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+```
+yarn smoke-tests --playwrightOptions="--debug"
+yarn smoke-tests --playwrightOptions="--headed"
+yarn smoke-tests --playwrightOptions="--ui"
+```
+
+See `yarn smoke-tests --help` for more.
+
+If you want to skip the CLI all together, you can `cd` into the directory of the smoke test you want to run and run `npx playwright test` directly:
```
Comment (Tobbe) [inline tasks/smoke-tests/README.md:40]: Maybe mention that you still need `REDWOOD_TEST_PROJECT_PATH` here when running `npx playwright test`

Comment (jtoar) [reply tasks/smoke-tests/README.md:40]: See https://github.com/redwoodjs/redwood/pull/9381/commits/2e48d81ca334fe627759b4fe25fdd7e294fd1aa1

**Pull request merged**